### PR TITLE
Support for mj-include

### PR DESCRIPTION
**Pull request by SebastianStehle**

### CODE REVIEW

**Comment by SebastianStehle** [comment]
@LiamRiddell I made a few additional changes:

1. The bind attribute accepts a secondary "type"-argument now to create an IType in the generated code.
2. I added additional arguments to IType and IValidator. To make further changes easier I introduced a ValidationContext struct.
3. I added the methods for xml fragment rendering for IncludeComponent (mj-include).

**Comment by LiamRiddell** [comment]
> @LiamRiddell I made a few additional changes:
> 
> 1. The bind attribute accepts a secondary "type"-argument now to create an IType in the generated code.
> 2. I added additional arguments to IType and IValidator. To make further changes easier I introduced a ValidationContext struct.
> 3. I added the methods for xml fragment rendering for IncludeComponent (mj-include).

Nice implementation, I looked through MJML docs and it seems when including `mjml` template the root `<mjml>` element is optional. From my understanding in the code it's required, so we'd need to add this as a note against the feature.

Apologies, I'm unable to pick-up anything been really busy on other projects. However, it looks like most of the issues have been ironed out. Thank you for the continued work! 👍🏼

**Comment by SebastianStehle** [comment]
> Nice implementation, I looked through MJML docs and it seems when including mjml template the root <mjml> element is optional. From my understanding in the code it's required, so we'd need to add this as a note against the feature.

I think the examples are weird. The resulting mjml (after all includes have been resolved), must still be valid.

> Apologies, I'm unable to pick-up anything been really busy on other projects. However, it looks like most of the issues have been ironed out. Thank you for the continued work! 👍🏼

Thanks. Something publicly available?

File: Mjml.Net/MjmlOptions.cs [hunk 1]
```diff
@@ -83,5 +83,10 @@ public sealed record MjmlOptions
         /// The current validator.
         /// </summary>
         public IValidatorFactory? ValidatorFactory { get; init; }
+
+        /// <summary>
+        /// The loader for included texts.
```
Comment (LiamRiddell) [inline Mjml.Net/MjmlOptions.cs:88]: @SebastianStehle, maybe the summary could be changed to `The file path loader for <mj-include path="..." type="..."> which handles loading the files from the specified path attribute. `

**Review by LiamRiddell — CHANGES_REQUESTED** [review]

File: Mjml.Net/MjmlRenderContext.cs [hunk 2]
```diff
@@ -4,14 +4,17 @@
 
 namespace Mjml.Net
 {
-    public sealed partial class MjmlRenderContext
+    public sealed partial class MjmlRenderContext : IMjmlReader
     {
+        private static readonly XmlReaderSettings ReaderSettings = new XmlReaderSettings { ConformanceLevel = ConformanceLevel.Fragment };
```
[the remark this answers is not in the document]
Comment (LiamRiddell) [reply Mjml.Net/MjmlRenderContext.cs:9]: I'll add this in new PR

**Comment by LiamRiddell** [comment]
Ignore above I'll make changed in new PR. 😄

**Pull request merged**